### PR TITLE
[FIX] Do not call _recompute_balance in the tests, but set the call i…

### PR DESCRIPTION
**Pull request by legalsylvain**
…n the correct place to make the field recomputed correctly

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

File: account_customer_wallet/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _post(self, soft=True):
+        res = super()._post(soft=soft)
+        if not soft:
+            partners = self.mapped("line_ids.partner_id")
+            partners |= partners.mapped("child_ids")
+            partners |= partners.mapped("parent_id")
```
Comment (carmenbianca) [inline account_customer_wallet/models/account_move.py]: Both of these should be recursive.

**Review by carmenbianca — COMMENTED** [review]
I like this PR. I have one remark below.

However, couldn't we do the "proper" thing and do something like:

- add One2many relation to account.move.line on partner (does another OCA module already do this?). The Many2one already exists, so this should be trivial.
- Make `_compute_customer_wallet_balance` depend on `move_lines`, `move_lines.parent_state`, `move_lines.balance`, `move_lines.account_id`, `parent_id`, and `child_ids` (maybe more?)

But maybe the dependency on `parent_id` and `child_ids` runs into the same recursivity problem as remarked below.

I'm also not certain whether it's possible to _append_ to `api.depends` when overriding a compute function (e.g. in pos_customer_wallet). Do you need to list all the depends again?

File: account_customer_wallet/models/account_move.py (hunk 1 above)
Comment (legalsylvain) [reply account_customer_wallet/models/account_move.py]: Done in the last commit. replaced by get_all_partners_in_family

**Comment by legalsylvain** [comment]
> I'm also not certain whether it's possible to append to api.depends when overriding a compute function (e.g. in pos_customer_wallet). Do you need to list all the depends again?

It's possible to write ``@api.depends(lambda self: self._customer_wallet_balance_depends())``
see : https://github.com/odoo/odoo/blob/16.0/odoo/addons/base/models/res_partner.py#L396



> Make _compute_customer_wallet_balance depend on move_lines, move_lines.parent_state, move_lines.balance, move_lines.account_id, parent_id, and child_ids (maybe more?)

Should depends on parent_id.customer_wallet_balance and child_ids.customer_wallet_balance ?

> However, couldn't we do the "proper" thing and do something like:

don't know. Do you want to make a try ? 

For my part, I would like to have an answer to this issue ;-) https://github.com/odoo/odoo/issues/104202

**Comment by carmenbianca** [comment]
> don't know. Do you want to make a try ?

I will give this a spin.

**Comment by carmenbianca** [comment]
Gave it a spin. It works! See recent commits to #290

**Pull request closed without merging**